calliope/test/test_core_future_warnings.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+
+import calliope
+from calliope.test.common.util import build_test_model as build_model
+
+
+@pytest.mark.filterwarnings('always::FutureWarning')
+class TestFutureWarnings:
+
+    def test_group_share_warning(self):
+        override = {
+            'model.group_share.test_supply_elec.carrier_prod_min.electricity': 0.85
+        }
+        expected_warning_msg = '`group_share` constraints will be removed in v0.7.0'
+        with pytest.warns(FutureWarning, match=expected_warning_msg):
```

Comment:
I removed the version check from this test and the following. I believe it's fine like that: should the warning (and the feature) be removed, this test will fail. We should then remove the test, because we are not expecting anything really. But maybe I am missing something here.

Reply:
Sounds sensible to me